packages/framework/src/Model/Statistics/StatisticsRepository.php

```diff
@@ -83,4 +86,76 @@ function (array $item) {
             $query->getResult()
         );
     }
+
+    /**
+     * @param \DateTimeImmutable $startDateTime
+     * @param \DateTimeImmutable $endDateTime
+     * @return int
+     */
+    public function getNewCustomersCountBetweenDates(DateTimeImmutable $startDateTime, DateTimeImmutable $endDateTime)
+    {
+        $resultSetMapping = new ResultSetMapping();
+        $resultSetMapping->addScalarResult('count', 'count');
+
+        $query = $this->em->createNativeQuery(
+            'SELECT COUNT(u.created_at) AS count
+            FROM users u
+            WHERE u.created_at BETWEEN :start_date AND :end_date',
+            $resultSetMapping
+        );
+
+        $query->setParameter('start_date', $startDateTime);
+        $query->setParameter('end_date', $endDateTime);
+
+        return (int)$query->getSingleScalarResult();
+    }
+
+    /**
+     * @param \DateTimeImmutable $startDateTime
+     * @param \DateTimeImmutable $endDateTime
+     * @return int
+     */
+    public function getOrdersCountBetweenDates(DateTimeImmutable $startDateTime, DateTimeImmutable $endDateTime)
+    {
+        $resultSetMapping = new ResultSetMapping();
+        $resultSetMapping->addScalarResult('count', 'count');
+
+        $query = $this->em->createNativeQuery(
+            'SELECT COUNT(o.created_at) AS count
+            FROM orders o
+            WHERE o.created_at BETWEEN :start_date AND :end_date AND o.status_id != :canceled',
+            $resultSetMapping
+        );
+
+        $query->setParameter('start_date', $startDateTime);
+        $query->setParameter('end_date', $endDateTime);
+        $query->setParameter('canceled', OrderStatus::TYPE_CANCELED);
+
+        return (int)$query->getSingleScalarResult();
+    }
+
+    /**
+     * @param \DateTimeImmutable $startDateTime
+     * @param \DateTimeImmutable $endDateTime
+     * @return int
+     */
+    public function getOrdersValueBetweenDates(DateTimeImmutable $startDateTime, DateTimeImmutable $endDateTime)
+    {
+        $resultSetMapping = new ResultSetMapping();
+        $resultSetMapping->addScalarResult('total_price', 'total_price');
+
+        $query = $this->em->createNativeQuery(
```

Comment:
is it better to use NativeQueries for this?

Reply:
Well yeah because you do not need any magic of entities, you just need to select raw data.